src/pudl/transform/eia923.py

```diff
@@ -1223,3 +1224,126 @@ def _core_eia923__fuel_receipts_costs(
+    csi_df.loc[
+        cfs_years_mask, csi_df.columns.str.endswith("_cubic_feet_per_second")
+    ] *= cfs_in_gpm
+    csi_df.columns = csi_df.columns.str.replace(
+        "_cubic_feet_per_second", "_gallons_per_minute"
+    )
+
+    # convert 1000 lbs to lbs
+    csi_df.loc[:, csi_df.columns.str.endswith("_1000_lbs")] *= 1000
+    csi_df.columns = csi_df.columns.str.replace("_1000_lbs", "_lbs")
+
+    # convert million gals to gals
+    csi_df.loc[:, csi_df.columns.str.endswith("_million_gallons")] *= 1_000_000
+    csi_df.columns = csi_df.columns.str.replace("_million_gallons", "_gallons")
+
+    return csi_df.pipe(apply_pudl_dtypes, group="eia", strict=False)
+
+
+@asset_check(asset=_core_eia923__cooling_system_information, blocking=True)
+def cooling_system_information_null_check(csi):
+    """We do not expect any columns to be completely null."""
+    pudl.validate.no_null_cols(csi)
+    return AssetCheckResult(passed=True)
+
+
+@asset_check(asset=_core_eia923__cooling_system_information, blocking=True)
+def cooling_system_information_withdrawal_discrepancy_check(csi):
+    """Withdrawal should be equal to discharge + consumption.
+
+    To allow for *some* data quality errors we assert that withdrawal ~=
+    discharge + consumption at least 90% of the time (with a 1% acceptable
+    discrepancy).
+    """
+
+    def sum_to_target_rate(sum_cols, target, threshold):
+        discrepancies = (
+            csi.loc[:, sum_cols].sum(axis="columns") - csi.loc[:, target]
+        ).dropna() / csi.loc[:, target]
+        return (discrepancies > threshold).sum() / len(discrepancies)
+
+    monthly_withdrawal_discrepancy_rate = sum_to_target_rate(
+        sum_cols=[
+            "monthly_average_consumption_rate_gallons_per_minute",
+            "monthly_average_discharge_rate_gallons_per_minute",
+        ],
+        target="monthly_average_withdrawal_rate_gallons_per_minute",
+        threshold=0.01,
+    )
+    assert monthly_withdrawal_discrepancy_rate < 0.1
+
+    annual_withdrawal_discrepancy_rate = sum_to_target_rate(
+        sum_cols=[
+            "annual_average_consumption_rate_gallons_per_minute",
+            "annual_average_discharge_rate_gallons_per_minute",
+        ],
+        target="annual_average_withdrawal_rate_gallons_per_minute",
+        threshold=0.01,
+    )
+    assert annual_withdrawal_discrepancy_rate < 0.1
+
+    return AssetCheckResult(passed=True)
+
+
+@asset_check(asset=_core_eia923__cooling_system_information, blocking=True)
+def cooling_system_information_continuity(csi):
+    """Check to see if columns vary as slowly as expected."""
+    return pudl.validate.group_mean_continuity_check(
+        df=csi,
+        thresholds={
+            "annual_average_consumption_rate_gallons_per_minute": 0.1,
+            "annual_average_discharge_rate_gallons_per_minute": 0.1,
+            "annual_average_withdrawal_rate_gallons_per_minute": 0.1,
+            "monthly_average_consumption_rate_gallons_per_minute": 0.5,
```

Comment:
I wonder if we want to up `n_outliers_allowed` and drop this down as it seems super high, but that's probably out of scope for this PR.

Reply:
I took a look at the `top` values for this and it was something like 1.4/0.49/0.48/0.46 etc - I think probably there's just pretty large month-to-month variation, but we could dig deeper into this once we have more time.